core/chains/orm.go

```diff
@@ -12,37 +12,81 @@ import (
 	"github.com/smartcontractkit/chainlink/core/services/pg"
 )
 
-// Chain is a generic DB chain for any configuration type.
-// CFG normally implements sql.Scanner and driver.Valuer, but that is not enforced here.
+// ORM manages chains and nodes.
+//
+// I: Chain ID
+// C: Chain Config
+// N: Node including these default fields:
+//  ID        int32
+//  Name      string
+//  CreatedAt time.Time
+//  UpdatedAt time.Time
+type ORM[I, C, N any] interface {
```

Comment:
Using stricter types than `any` would require pretty broad refactoring, and didn't seem worth it for something that will likely be phased out. But FWIW the exported API surface was reduced and simplified a lot :shrug:

Reply:
godoc:
![Screenshot from 2022-04-19 14-12-06](https://user-images.githubusercontent.com/1194128/164078375-6e599792-b7eb-4aff-ae2b-55265cd4a3b9.png)